### DISCUSSION TITLE
S22 or S14?

### DISCUSSION THREAD

**Discussion by Lieta2** [Benas09/Blauberg_S14#1] [General]
Hello.
According to description of [KOMFORT ERV EC DB 350](https://blaubergventilatoren.de/en/product/komfort-erv-ec-db350-s14#description) it supports S14 automation functions, not S22. Could you, please, clarify?

**Comment by Benas09** [comment]
Hello, good question. These two controllers looks the same in photos, so I wrote that I have S22. But by recuperator unit datasheet it should be S14. I have one spare controller, so I will try to dissasemble it and search written name or something on it.

**Comment by Benas09** [comment]
I didn't find any markings for S14 or S22. So, as according to the description, it should be S14. 

I renamed in code and also added defrosting operation (turn off fans for 2 minutes when received status is 0x02.

![1704201708001](https://github.com/Benas09/Blauberg_S22/assets/29705152/3e9b7ff9-e88d-4639-ba42-7c914f6b74ac)
![1704201708011](https://github.com/Benas09/Blauberg_S22/assets/29705152/48cf1305-ec5e-4934-abc4-9ec908ca544c)
![1704201708022](https://github.com/Benas09/Blauberg_S22/assets/29705152/3008c2ce-a9c5-418e-a09a-c5ea720a925f)

![image](https://github.com/Benas09/Blauberg_S22/assets/29705152/47b117c2-9dd9-4969-8716-9463b18f7982)

**Comment by matkvaid** [comment]
Hi, 
sorry for stupid questions - no experience with such boards, but i want to add same blauberg model to HA, am i understanding this correct:
so i would need these two and connect 24v and gnd to voltage converter from unit, then tx and rx directly to d1 board? 

<img width="868" alt="Screenshot 2024-08-19 at 22 04 23" src="https://github.com/user-attachments/assets/d2d841c8-9549-4b73-b1c4-7b24ccc8d9f1">
<img width="880" alt="Screenshot 2024-08-19 at 22 04 13" src="https://github.com/user-attachments/assets/ce521f08-5a13-4098-89a8-38f89e707fe8">

**Comment by Benas09** [comment]
Hello. Your given power converter is overkill (it states it is capable 75W). I use this with no problems
![image](https://github.com/user-attachments/assets/75d12d4d-c2a4-48ac-ae18-15bf1427e0ed)

I updated readme file with circuit diagram.

**Comment by matkvaid** [comment]
Seems this power converter solved at least hardware watchdog restart issue. AP63203 connected to esp board 3.3V pin, i do not have hardware watchdog restart for 3 days
![ap63203-buck-laiptelio-keitiklis-su-varzto-jungtimi-3-3-v-2a-sparkfun-com-18356-219262-500x500](https://github.com/user-attachments/assets/8d646c28-6f3a-428d-b877-6c2fad2d398e)

**Reply by Benas09** [reply]
Still no restarts after replacing d/DC converter?

**Reply by matkvaid** [reply]
I did not have restarts with hardware watchdog reason for some time, but it still restarts, random reasons again. I found out that these are Wemos clones with some bad LDO that has only 150mA instead of 500mA (not sure if this LDO do something when supplied with 3.3V directly) - https://github.com/bbqkees/Wemos-the-Clone-Wars. With this dc/dc i now have ±2.93 to 2.95V voltage reported by VCC sensor, will try usb power tomorrow too. And ordered new module directly from official Wemos aliexpress store, will check maybe these are better. I also had chatgpt help to make in yaml fan with 3 speed switches instead of 4 separate switches for speeds, can share

**Reply by Benas09** [reply]
I think, if pass 3V directly, you bypass internal voltage regulator and it is not used anymore. Then maybe restarts may be due to weak wifi signal? Mine controller is hidden in the closet behind mirror door.
It would be nice if you share 3 speed switch config. 

Also today I added filter reset feature

**Comment by matkvaid** [comment]
Wifi is ~-62db, mine is just hidden under glass panel (removed original controller and put esp board in wall box). This is also with diagnostic sensors and i also now use mqtt, not esphome api, but not much difference. Will share experience when not cloned board will arrive. 

With this one i had to change automations to percentage, like set speed to 33/66/100%. and as i understand - now it sees is as "one device", you need to "turn on" first, so the automation after restart must first "turn on device", only then i can set speed or activate previous scene:

uart:
  id: uart_bus
  tx_pin: D0
  rx_pin: D1
  baud_rate: 600
  debug:
    direction: BOTH
    dummy_receiver: false
    after:
      delimiter: "\n"
    sequence:
      - lambda: |-
          UARTDebug::log_string(direction, bytes);  //Still log the data


globals:
  - id: fan_speed
    type: int
    restore_value: yes
    initial_value: '1'

custom_component:
  - lambda: |-
      auto blauberg = ${init};
      return {blauberg};

output:
  - platform: template
    id: blauberg_fan_output
    type: float
    write_action:
      lambda: |-
        auto blauberg = ${init};
        if (state <= 0.25) {
          blauberg->set_speed(0);
          id(fan_speed) = 0;
        } else if (state <= 0.5) {
          blauberg->set_speed(1);
          id(fan_speed) = 1;
        } else if (state <= 0.75) {
          blauberg->set_speed(2);
          id(fan_speed) = 2;
        } else {
          blauberg->set_speed(3);
          id(fan_speed) = 3;
        }

fan:
  - platform: speed
    output: blauberg_fan_output
    name: "${name}_Fan"
    speed_count: 3

switch:
  - platform: template
    name: "${name}_Damper"
    icon: mdi:hvac
    lambda: |-
      auto blauberg = ${init};
      return blauberg->get_damper();
    turn_on_action:
      lambda: |-
        auto blauberg = ${init};
        blauberg->set_damper(true);
    turn_off_action:
      lambda: |-
        auto blauberg = ${init};
        blauberg->set_damper(false);

  - platform: restart
    name: "${name}_Restart"

sensor:
  - platform: custom
    lambda: |-
      auto blauberg = ${init};
      return {
        blauberg->sensor_response,
        blauberg->sensor_speed
      };
    sensors:
      - name: "${name}_Response"
        icon: mdi:cable-data
      - name: "${name}_Speed"
        icon: mdi:fan
  - platform: wifi_signal
    name: "WiFi Signal Sensor"
    update_interval: 60s
  - platform: uptime
    name: "${friendly_name} Uptime"
    filters:
      - lambda: return x / 60.0;
    unit_of_measurement: minutes
    entity_category: "diagnostic"
  - platform: debug
    free:
      name: "${friendly_name} Heap Free"
    fragmentation:
      name: "${friendly_name} Heap Fragmentation"
    block:
      name: "${friendly_name} Heap Max Block"
    loop_time:
      name: "${friendly_name} Loop Time"
  - platform: adc
    pin: VCC
    name: "${friendly_name} VCC"
    update_interval: 5s


text_sensor:
  - platform: debug
    reset_reason:
      name: "${friendly_name} Reset Reason"
    device:
      name: "${friendly_name} Device Info"

binary_sensor:
  - platform: status
    name: "${friendly_name} Status"
    entity_category: "diagnostic"
  - platform: custom
    lambda: |-
      auto blauberg = ${init};
      return {
        blauberg->sensor_damper,
        blauberg->sensor_isDefrosting
      };
    binary_sensors:
      - name: "${name}_Damper"
        device_class: opening
        icon: mdi:hvac
      - name: "${name}_Defrosting"
        device_class: running
        icon: mdi:snowflake-melt

debug:
  update_interval: 5s

**Comment by matkvaid** [comment]
Current update about restart issues: with Wemos S2 Mini (from Ali official store), no restarts for over 40 hours already :D Also logs does not show any errors (with d1 board there was some issue about incorrect loop time).

**Comment by Benas09** [comment]
How about restarts after a month?

**Reply by matkvaid** [reply]
No more random restarts few times a day, had ~14 days uninterupted max, restarts when wifi looses connection or i power off manually

**Reply by Benas09** [reply]
O changed UART pins to GPIO1 and GPIO3, to use hardware serial instead of software serial and restart issue is solved.
Also, serial logger must be disabled to prevent sending logs data to recuperator unit.
